### PR TITLE
Syntax error on a break outside any loop

### PR DESCRIPTION
**Pull request by stevelinton**
This PR adds code to detect when a break or continue statement is being parsed (technically coded) and is not contained in a loop within the current function body.

### CODE REVIEW

**Comment by ChrisJefferson** [comment]
This code looks fine, and seems to work. While convincing myself it was right, I wrote a little test ( on branch pr/304 on ChrisJefferson). Feel free to grab testinstall/break.tst from there.

Discovered repeat -- until loops. Didn't know GAP had those (they work fine)

**Comment by ChrisJefferson** [comment]
Woops, my fault, need a 'local i' in some of the inside functions. Sorry.

File: src/read.c [hunk 1]
```diff
@@ -1251,6 +1251,7 @@ void ReadFuncExpr (
     /* 'function( ... arg )' takes a variable number of arguments              */
     if (narg >= 1 && ! strcmp( "arg", CSTR_STRING( ELM_LIST(nams, narg) ) ) )
       {
+	/* TODO: remove this before the next non-beta release */
```
Comment (fingolfin) [inline src/read.c]: This change seems to be unrelated, so perhaps it should not be in this PR?
It also seems fragile. Perhaps we should have a GAP_BETA `#define`, and then the above could become
```
#if !defined(GAP_BETA)
#error Please revise this code before making a non-beta release
#endif
```

**Comment by fingolfin** [comment]
Overall I am quite happy with this, and think it should be merged -- except perhaps the first commit should be dropped, and the fixup commit for break.tst should be squashed into the commit before it (both trivial to do with a interactive rebase on master, i.e. `git rebase -i master`).

But I'll survive if this is not done, too ;-)

File: src/read.c (hunk 1 above)
Comment (stevelinton) [inline src/read.c]: You're right. Sorry unconnected change got caught up.

I'm not sure how common this kind of thing is. I guess extra warnings, or warnings instead of errors in the beta release might be fairly normal in which case this makes sense.

**Comment by fingolfin** [comment]
@stevelinton You do not need to close a PR in order to make changes to the code -- you can simply update the branch, and push it, and the PR will automatically be updated. :)

**Comment by stevelinton** [comment]
For whatever reason I was not able to get the rebase to propagate correctly, so I built a new branch.
